tests/phpunit/testcases/test_template_helper_functions.php

```diff
@@ -549,4 +538,1313 @@ function test_map_glossary_entries_to_translation_originals_with_suffixes( $test
+				'translation' => 'preceder', // Portuguese.
+				'glossary_id' => null,
+			),
+			array(
+				'term' => 'decide', // Decide and decision.
+				'part_of_speech' => $part_of_speech,
+				'translation' => 'decidir', // Portuguese.
+				'glossary_id' => null,
+			),
+			array(
+				'term' => 'explode', // Explode and explosion.
+				'part_of_speech' => $part_of_speech,
+				'translation' => 'explodir', // Portuguese.
+				'glossary_id' => null,
+			),
+			array(
+				'term' => 'exclude', // Exclude and exclusion.
+				'part_of_speech' => $part_of_speech,
+				'translation' => 'excluir', // Portuguese.
+				'glossary_id' => null,
+			),
+			array(
+				'term' => 'supervise', // Supervise and supervision.
+				'part_of_speech' => $part_of_speech,
+				'translation' => 'supervisionar', // Portuguese.
+				'glossary_id' => null,
+			),
+			array(
+				'term' => 'confuse', // Confuse and confusion.
+				'part_of_speech' => $part_of_speech,
+				'translation' => 'confundir', // Portuguese.
+				'glossary_id' => null,
+			),
+			array(
+				'term' => 'expel', // Expel and expulsion.
+				'part_of_speech' => $part_of_speech,
+				'translation' => 'expulsar', // Portuguese.
+				'glossary_id' => null,
+			),
+			array(
+				'term' => 'submit', // Submit and submission.
+				'part_of_speech' => $part_of_speech,
+				'translation' => 'submeter', // Portuguese.
+				'glossary_id' => null,
+			),
+			array(
+				'term' => 'compress', // Compress and compression.
+				'part_of_speech' => $part_of_speech,
+				'translation' => 'comprimir', // Portuguese.
+				'glossary_id' => null,
+			),
+			array(
+				'term' => 'extend', // Extend and extension.
+				'part_of_speech' => $part_of_speech,
+				'translation' => 'estender', // Portuguese.
+				'glossary_id' => null,
+			),
+			array(
+				'term' => 'convert', // Convert and conversion.
+				'part_of_speech' => $part_of_speech,
+				'translation' => 'converter', // Portuguese.
+				'glossary_id' => null,
+			),
+			array(
+				'term' => 'disperse', // Disperse and dispersion.
+				'part_of_speech' => $part_of_speech,
+				'translation' => 'dispersar', // Portuguese.
+				'glossary_id' => null,
+			),
+			array(
+				'term' => 'recur', // Recur and recursion.
+				'part_of_speech' => $part_of_speech,
+				'translation' => 'repetir', // Portuguese.
+				'glossary_id' => null,
+			),
+			array(
+				'term' => 'emerge', // Emerge and emersion
+				'part_of_speech' => $part_of_speech,
+				'translation' => 'emergir', // Portuguese.
+				'glossary_id' => null,
+			),
+
+		);
+
+		$this->check_map_glossary( $test_string, $part_of_speech, $matches, $glossary_entries );
+	}
+
+	/**
+	 * Method to test the map_glossary_entries_to_translation_originals() function.
+	 *
+	 * @param $test_string      The string to test.
+	 * @param $part_of_speech   The part of speech of the string to test.
+	 * @param $matches          The matches to expect.
+	 * @param $glossary_entries The glossary entries to create.
```

Comment:
One small change in the header of the `check_map_glossary` method: I have added the type of each parameter, so the IDE doesn't show any warning and we can use it in the documentation and in another similar places: 

```suggestion
	 * @param $test_string      string The string to test.
	 * @param $part_of_speech   string The part of speech of the string to test.
	 * @param $matches          array  The matches to expect.
	 * @param $glossary_entries array  The glossary entries to create.
```

Reply:
You're right, missed the types.
Still, it should be before the variable:
@param string $var

Right?